apps/pay/app/[username]/layout.tsx

```diff
@@ -0,0 +1,15 @@
+import React from "react"
+
+import AppLayout from "@/components/Layouts/AppLayout"
+
+export default function UserNameLayout({
+  children,
+  params,
+}: {
+  children: React.ReactNode
+  params: {
+    username: string
+  }
+}) {
+  return <AppLayout username={params.username}>{children}</AppLayout>
```

Comment:
are we making proper use of `Layout` here?

Layour are nested, so if we need an app layout, we should just have it at the root and we don't need to reimport an `AppLayout` further down the tree of the router

Reply:
Maybe let's call it something else than `AppLayout`. `UsernameLayout` would be more appropriate